tagstudio/src/qt/widgets/migration_modal.py

```diff
@@ -0,0 +1,785 @@
+
+    def check_ext_type(self) -> bool:
+        return self.json_lib.is_exclude_list == self.sql_lib.prefs(LibraryPrefs.IS_EXCLUDE_LIST)
+
+    def check_alias_parity(self) -> bool:
+        """Check if all JSON aliases match the new SQL aliases."""
+        sql_aliases: set[str] = None
+        json_aliases: set[str] = None
+
+        with Session(self.sql_lib.engine) as session:
+            for tag in self.sql_lib.tags:
+                tag_id = tag.id  # Tag IDs start at 0
+                sql_aliases = set(
+                    session.scalars(select(TagAlias.name).where(TagAlias.tag_id == tag.id))
+                )
+                json_aliases = set(self.json_lib.get_tag(tag_id).aliases)
+
+                logger.info(
+                    "[Alias Parity]",
+                    tag_id=tag_id,
+                    json_aliases=json_aliases,
+                    sql_aliases=sql_aliases,
+                )
+                if not (
+                    sql_aliases is not None
+                    and json_aliases is not None
+                    and (sql_aliases == json_aliases)
+                ):
+                    self.discrepancies.append(
+                        f"[Alias Parity]:\nOLD (JSON):{json_aliases}\nNEW (SQL):{sql_aliases}"
+                    )
+                    self.alias_parity = False
+                    return self.alias_parity
+
+        self.alias_parity = True
+        return self.alias_parity
+
+    def check_shorthand_parity(self) -> bool:
+        """Check if all JSON shorthands match the new SQL shorthands."""
+        sql_shorthand: str = None
+        json_shorthand: str = None
+
+        for tag in self.sql_lib.tags:
+            tag_id = tag.id  # Tag IDs start at 0
+            sql_shorthand = tag.shorthand
+            json_shorthand = self.json_lib.get_tag(tag_id).shorthand
+
+            logger.info(
+                "[Shorthand Parity]",
+                tag_id=tag_id,
+                json_shorthand=json_shorthand,
+                sql_shorthand=sql_shorthand,
+            )
+
+            if not (
+                sql_shorthand is not None
+                and json_shorthand is not None
+                and (sql_shorthand == json_shorthand)
+            ):
+                self.discrepancies.append(
+                    f"[Shorthand Parity]:\nOLD (JSON):{json_shorthand}\nNEW (SQL):{sql_shorthand}"
+                )
+                self.shorthand_parity = False
+                return self.shorthand_parity
+
+        self.shorthand_parity = True
+        return self.shorthand_parity
+
+    def check_color_parity(self) -> bool:
+        """Check if all JSON tag colors match the new SQL tag colors."""
+        sql_color: str = None
+        json_color: str = None
+
+        for tag in self.sql_lib.tags:
+            tag_id = tag.id  # Tag IDs start at 0
+            sql_color = tag.color.name
+            json_color = (
+                self.json_lib.get_tag(tag_id).color.upper().replace(" ", "_")
```

Comment:
`.upper().replace(" ", "_")`

this conversion is duplicated in `TagColor.get_color_from_str` maybe that should be a static method of `TagColor`?